docs/users/step-debugging.md

```diff
@@ -93,26 +93,61 @@ xdebug.client_port=9000
 
 * Then change your IDE's configuration to listen on the new port.
 
-NOTE: If you are using a PHP version below PHP7.2, you will be using Xdebug version 2.x, instead of 3.x. In that case the port config should be `xdebug.remote_port` instead.
+NOTE: If you are using a PHP version below PHP7.2, you will be using Xdebug version 2.x, instead of 3.x. In that case
+the port config should be `xdebug.remote_port` instead.
 
 ### Troubleshooting Xdebug
 
-The basic thing to understand about xdebug is that it's a network protocol. Your IDE (like PhpStorm) will listen on the xdebug port (9003 by default in v1.19+, previously 9000). Then if xdebug is enabled in the ddev web container with `ddev xdebug on`, then php inside the container will try to open a TCP connection to the IDE. Docker's networking places the host-side listening IDE at `host.docker.internal:9003`. So you have to make sure that the network connection is clear and can be made and everything should work. Here are basic steps to take to sort out any difficulty:
+xdebug is a server-side tool: It is installed automatically on the container and you do *not* need to install or
+configure it on your workstation.
+
+The basic thing to understand about xdebug is that it's a network protocol. Your IDE (like PhpStorm) will listen on the
+xdebug port (9003 by default in v1.19+, previously 9000). Then if a xdebug session is started, php inside the container
+will try to open a TCP connection to the IDE. When starting the session via a browser xdebug will try to connect back to
+client, otherwise `host.docker.internal:9003` will be used to connect, which is provided by docker. So you have to make
```

Comment:
```suggestion
client, otherwise `host.docker.internal:9003` will be used to connect, which is provided by DDEV or Docker Desktop. So you have to make
```

Reply:
host.docker.internal is not set when using docker on linux?

Reply:
It's set by ddev by exploring the network topology on Linux or Colima. Only Docker Desktop sets it otherwise. Ddev generates docker-compose stanza to define it, and figures out what it should be. Not perfect either.